eclair-core/src/main/scala/fr/acinq/eclair/wire/Onion.scala

```diff
@@ -293,6 +293,9 @@ object Onion {
     FinalTlvPayload(TlvStream(AmountToForward(amount), OutgoingCltv(expiry), PaymentData(paymentSecret, totalAmount), TrampolineOnion(trampolinePacket)))
   }
 
+  def createCustomRecordsFinalPayload(amount: MilliSatoshi, expiry: CltvExpiry, additionalTlvRecords: Seq[GenericTlv] = Seq.empty): FinalPayload =
```

Comment:
I'm not sure we need a new function, I would just add the `additionalTlvRecords: Seq[GenericTlv] = Seq.empty` parameter to `createSinglePartPayload` and the two `createMultiPartPayload`.
For instance it makes sure we provide a `paymentSecret` which is really important even without MPP.

Reply:
:+1:  refactored and added the extra param to both methods